src/borg/repository.py

```diff
@@ -107,10 +107,14 @@ def __init__(self, id, repo):
     class InsufficientFreeSpaceError(Error):
         """Insufficient free space to complete transaction (required: {}, available: {})."""
 
-    def __init__(self, path, create=False, exclusive=False, lock_wait=None, lock=True, append_only=False):
+    class StorageQuotaExceeded(Error):
+        """The storage quota ({}) has been exceeded ({}). Try deleting some archives."""
```

Comment:
Actually that wasn't quite right, since the `.traceback` attributes of Error was not propagated this did not quite work as intended. Fixed below.